Server/src/RedisManager.cpp

```diff
@@ -0,0 +1,157 @@
+ * terms of the Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ */
+
+#include "RedisManager.h"
+
+
+/*********************************************************************//**
+ * Constructor for class
+ ***********************************************************************/
+RedisManager::RedisManager() : stateDb_(BMP_DB_NAME, 0, true) {
+    exit_ = false;
+}
+
+/*********************************************************************//**
+ * Constructor for class
+ ***********************************************************************/
+RedisManager::~RedisManager() {
+}
+
+
+/*********************************************************************
+ * Setup for this class
+ *
+ * \param [in] logPtr     logger pointer
+ * \param [in] cfgPtr     config pointer
+ ***********************************************************************/
+void RedisManager::Setup(Logger *logPtr, Config *cfgPtr) {
+    logger = logPtr;
+    if (!cfgPtr->redis_multiAsic) {
+        if (!swss::SonicDBConfig::isInit()) {
+            swss::SonicDBConfig::initialize();
+        }
+    } else {
+        if (!swss::SonicDBConfig::isGlobalInit()) {
+            swss::SonicDBConfig::initializeGlobalConfig();
+        }
+    }
+    separator_ = swss::SonicDBConfig::getSeparator(BMP_DB_NAME);
+}
+
+
+
+/**
+ * Get Key separator for deletion
+ *
+ * \param [in] N/A
+ */
+std::string RedisManager::GetKeySeparator() {
+    return separator_;
+}
+
+
+/**
+ * WriteBMPTable
+ *
+ * \param [in] table            Reference to table name
+ * \param [in] key              Reference to various keys list
+ * \param [in] fieldValues      Reference to field-value pairs
+ */
+bool RedisManager::WriteBMPTable(const std::string& table, const std::vector<std::string>& keys, const std::vector<swss::FieldValueTuple> fieldValues) {
+
+    if (enabledTables_.find(table) == enabledTables_.end()) {
+        LOG_INFO("RedisManager %s is disabled", table.c_str());
+        return false;
+    }
+
+    swss::Table stateBMPTable(&stateDb_, table);
+    std::string fullKey;
+    for (const auto& key : keys) {
+        fullKey += key;
+        fullKey += separator_;
+    }
+    fullKey.erase(fullKey.size() - 1);
+
+    LOG_INFO("RedisManager WriteBMPTable key = %s", fullKey.c_str());
+
+    stateBMPTable.set(fullKey, fieldValues);
+    return true;
+}
+
+
+/**
+ * RemoveEntityFromBMPTable
+ *
+ * \param [in] keys             Reference to various keys
+ */
+bool RedisManager::RemoveEntityFromBMPTable(const std::vector<std::string>& keys) {
+
+    stateDb_.del(keys);
+    return true;
+}
+
+
+/**
+ * ExitRedisManager
+ *
+ * \param [in] N/A
+ */
+void RedisManager::ExitRedisManager() {
+    exit_ = true;
+}
+
+
+/**
+ * InitBMPConfig, read config_db for table enablement setting.
+ *
+ * \param [in] N/A
+ */
+bool RedisManager::InitBMPConfig() {
+    auto connector = swss::ConfigDBConnector_Native();
```

Comment:
ConfigDBConnector_Native is for SWIG wrapper.

You need use DBConnector https://github.com/liuh-80/sonic-swss-common/blob/master/common/dbconnector.h

and Table: https://github.com/liuh-80/sonic-swss-common/blob/master/common/table.h#L189

Reply:
updated.

Reply:
#closed